indexer.ts

```diff
@@ -0,0 +1,743 @@
+    )
+    .map((receipt) => receipt.receiptId);
+
+  console.log("let us see...", receiptStatusMap);
+
+  // filter actions in this block whose receipts can be found in the list of successful receipts
+  // const matchingTx = block.transactions.filter(
+  //   (tx) => tx.receiverId.endsWith(BASE_ACCOUNT_ID) && tx.status
+  // );
+  // console.log(matchingTx);
+
+
+  // filter actions in this block whose receipts can be found in the list of successful receipts
+  const matchingActions = block
+    .actions()
+    .filter(
+      (action) =>
+        action.receiverId.endsWith(BASE_ACCOUNT_ID) &&
+        receiptStatusMap.includes(action.receiptId)
+    );
+
+  console.log("macthing actions rambo=====", matchingActions);
+
+  // if there are no actions matching our conditions, then we have nothing to index in this block, unto the next.
+  if (!matchingActions.length) {
+    console.log("nothin concerns us here");
+    return;
+  }
+
+  const functionCallTracked = [
+    "CREATE_ACCOUNT",
+    "deploy_pot",
+    "deploy_pot_callback",
+    "register",
+    "apply",
+    "assert_can_apply_callback",
+    "handle_apply",
+    "challenge_payouts",
+    "chef_set_payouts",
+  ];
+
+  async function handleNewPot(
+    args,
+    receiverId,
+    signerId,
+    predecessorId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new pot data::", { ...data }, receiverId, signerId);
+    await context.db.Account.upsert({ id: data.owner }, ["id"], []);
+    await context.db.Account.upsert({ id: signerId }, ["id"], []);
+
+    if (data.chef) {
+      await context.db.Account.upsert({ id: data.chef }, ["id"], []);
+    }
+
+    if (data.admins) {
+      for (const admin in data.admins) {
+        await context.db.Account.upsert({ id: admin }, ["id"], []);
+        let pot_admin = {
+          pot_id: receiverId,
+          admin_id: admin,
+        };
+        await context.db.PotAdmin.insert(pot_admin);
+      }
+    }
+
+    const deploy_time = new Date(
+      Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+    );
+
+    const potObject = {
+      id: receiverId,
+      pot_factory_id: predecessorId,
+      deployer_id: signerId,
+      deployed_at: deploy_time,
+      source_metadata: JSON.stringify(data.source_metadata),
+      owner_id: data.owner,
+      chef_id: data.chef,
+      name: data.pot_name,
+      description: data.pot_description,
+      max_approved_applicants: data.max_projects,
+      base_currency: null,
+      application_start: new Date(data.application_start_ms),
+      application_end: new Date(data.application_end_ms),
+      matching_round_start: new Date(data.public_round_start_ms),
+      matching_round_end: new Date(data.public_round_end_ms),
+      registry_provider: data.registry_provider,
+      min_matching_pool_donation_amount: data.min_matching_pool_donation_amount,
+      sybil_wrapper_provider: data.sybil_wrapper_provider,
+      custom_sybil_checks: data.custom_sybil_checks,
```

Comment:
`data.custom_sybil_checks` will be either `null` or an object, so if it is not null it should be stringified